examples/trk_subsampler.rs

```diff
@@ -40,18 +44,44 @@ fn main() {
     let mut writer = Writer::new(
         args.get_str("<output>"), Some(reader.header.clone())).unwrap();
 
+    let mut rng = match args.get_str("--seed").parse::<u8>() {
+        Ok(seed) => SmallRng::from_seed([seed; 16]),
+        _        => SmallRng::from_entropy()
+    };
+
     if let Ok(percent) = args.get_str("--percent").parse::<f32>() {
         let percent = percent / 100.0;
-        let mut rng = rand::thread_rng();
 
         for streamline in reader.into_iter() {
             if rng.gen::<f32>() < percent {
                 writer.write(&streamline);
             }
         }
-    } else if let Ok(_nb) = args.get_str("--number").parse::<usize>() {
-        panic!("Not implemented yet");
+    } else if let Ok(nb) = args.get_str("--number").parse::<usize>() {
+        let size = reader.header.nb_streamlines;
+        let number = size.min(nb);
+
+        if nb > size {
+            println!(
+                "The number {} exceed the total number of streamlines: {}. \
+                 Saving {} streamlines.",
+                nb, size, size);
```

Comment:
To have more standard messages, can you change to

    You requested a subsampling of {} streamlines, which is more than the total number of streamlines. The input file will simply be copied to the output file.

Moreover, when I wrote "optimize this", I was thinking about a simple read-write loop. Don't create, fill, shuffle and sort vector for nothing.

Also, this block (`--number`) is getting quite long, do you think it would be better to put it in its own function with a clear name?

Reply:
Is it fine if I do this?
```
if nb == 0 {
    panic!("message");
}
if nb > size {
    println!("message");
    for loop copy;
} else {
    call function sampling save;
}
```

Reply:
Yeah, something like that.